EIPS/returndatacopy.md

```diff
@@ -0,0 +1,68 @@
+
+Compiler implementors are advised to reserve a zero-length area for return data if the size of the return data is unknown before the call and then use `RETURNDATACOPY` in conjunction with `RETURNDATASIZE` to actually retrieve the data.
+
+Note that this proposal also makes the EIP that proposes to allow to return data in case of an intentional state reversion (EIP [206](https://github.com/ethereum/EIPs/pull/206)) much more useful. Since the size of the failure data might be larger than the regular return data (or even unknown), it is possible to retrieve the failure data after the CALL opcode has signalled a failure, even if the regular output area is not large enough to hold the data.
+
+## Specification
+
+Add two new opcodes and amend the semantics of any opcode that creates a new call frame (like `CALL`, `CREATE`, `DELEGATECALL`, ...) called call-like opcodes in the following. It is assumed that the EVM (to be more specific: an EVM call frame) has a new internal buffer of variable size, called the return data buffer. This buffer is created empty for each new call frame. Upon executing any call-like opcode, the buffer is cleared (its size is set to zero). After executing a call-like opcode, the complete return data (or failure data, see EIP [206](https://github.com/ethereum/EIPs/pull/206)) of the call is stored in the return data buffer (of the caller), and its size changed accordingly. As an exception, `CREATE` is considered to return the empty buffer in the success case and the failure data in the failure case. 
+
+As an optimization, it is possible to share the return data buffer across call frames because only one will be non-empty at any time.
+
+`RETURNDATASIZE`: `0x3d`
+
+Pushes the size of the return data buffer onto the stack.
+Gas costs: 2 (same as `CALLDATASIZE`)
+
+`RETURNDATACOPY`: `0x3e`
+
+This opcode has similar semantics to `CALLDATACOPY`, but instead of copying data from the call data, it copies data from the return data buffer. If the return data buffer is accessed beyond its size, it is considered to be filled with zeros.
+
+ALTERNATIVE: If the return data is accessed beyond its size, results in failure.
```

Comment:
A determination has been made in the [core developer call on 6/2/2017](https://youtu.be/E77sdcZZH0s?t=1184) that we will go ahead and go with the `throwing` approach rather than filling with zeros. This decision was made amongst the clients participating in the call, including geth and parity clients who re-iterated their stances and came to the resolution of using `throw` and pyethereum client who did not have an opinion on the subject.